src/compile/data/transforms.ts

```diff
@@ -130,11 +132,24 @@ export function parseTransformArray(model: Model) {
   let previous: DataFlowNode;
   let lookupCounter = 0;
 
+  function insert(newNode: DataFlowNode) {
+    if (!first) {
+      // A parent may be inserted during node construction
+      // (e.g., selection FilterNodes may add a TimeUnitNode).
+      first = newNode.parent || newNode;
```

Comment:
Because that would break encapsulation, I think. Right now, when the selection predicate is generated, [it inserts](https://github.com/vega/vega-lite/commit/96eee6705fbd20f15c5006cf6df2b0ca6406238c#diff-a91aafe62501d3fb2628cc5e5ab851efL223) the time unit into the dataflow. But see [my question](https://github.com/vega/vega-lite/commit/96eee6705fbd20f15c5006cf6df2b0ca6406238c#commitcomment-22981732) where I wonder whether this isn't the same logic as I had earlier, but just encapsulated in the `insertAsParentOf` method.